python/cudf/cudf/utils/dtypes.py

```diff
@@ -331,6 +331,8 @@ def to_cudf_compatible_scalar(val, dtype=None):
 
     If `val` is None, returns None.
     """
+    from cudf._lib.scalar import _is_null_host_scalar
```

Comment:
Why did this move to a function local import here?

Reply:
It's a circular import issue - I think it can be fixed by just moving this function, but then this PR would have to update everywhere in the python layer that imports it from here which is sort of out of scope. I'll raise an issue.

Reply:
It was to avoid this compilation error:
```
python/cudf/cudf/utils/dtypes.py:16: in <module>
    from cudf._lib.scalar import DeviceScalar, _is_null_host_scalar
E   ImportError: cannot import name '_is_null_host_scalar' from 'cudf._lib.scalar' (/home/cph/dev/rapids/cudf/python/cudf/cudf/_lib/scalar.cpython-37m-x86_64-linux-gnu.so)
```
Let me know if there is a better fix.

Reply:
Could we just replace the usage here with an absolute path? i.e. instead of calling `_is_null_host_scalar` we can use `cudf._lib.scalar._is_null_host_scalar`.

The reason I harp on this is that when doing imports inside of functions like this it goes through a list handler to check the modules and that list handler is surprisingly slow.

Reply:
specifically it won't compile without this change because this PR makes it so that `cudf._lib.types` imports stuff from `cudf.utils.dtypes` where `cudf.utils.dtypes` already imports from `cudf._lib.types`. One needs to not import from the other. They're being imported to make `dtype_to_data_type` to work, so either `cudf.utils.dtypes` needs to not import from here or this function would need to move to another file.

Reply:
Or, the runtime import of course.

Reply:
I'm fine with whatever approach is the easiest / best to fix this, but I'd really prefer not to have the function local import here because we pay the list handling cost at every invocation.

Reply:
It seems like unless we want to make some unrelated changes in the python layer part of this PR, we are going to need to do something temporary here and then fix it in a follow up. To that end I think the easiest thing would be to just leave this as is, merge it and then move it back out in a PR that also removes the dependency of `cudf.utils.dtypes` on `cudf._lib.types`.

Reply:
does removing the import and changing `_is_null_host_scalar` to `cudf._lib.scalar._is_null_host_scalar` not resolve the issue?

Reply:
It technically fixes the problem but I think we still probably don't want the circular dependency of `cudf.utils.dtypes` importing from `cudf._lib.scalar`, which imports from `cudf._lib.types`, which as of this PR would than in turn import from `cudf.utils.dtypes`.

Reply:
Yes I agree we should solve this problem, but using the absolute path resolves the performance issue for now and allows us to tackle this in a follow up PR.